api/api/admin/__init__.py

```diff
@@ -47,32 +57,214 @@ class MediaReportAdmin(admin.ModelAdmin):
+            return title
+
+        if obj.media_obj.creator and obj.media_obj.creator_url:
+            return mark_safe(
+                f"{title} by {create_link(obj.media_obj.creator_url, obj.media_obj.creator)}"
+            )
+        else:
+            creator = obj.media_obj.creator or obj.media_obj.creator_url
+            creator_string = f" by {creator}" if creator else ""
+            return f"{title}{creator_string}"
+
+    def media_display(self, obj):
+        pass
+
+    @admin.display(description="Media description")
+    def media_description(self, obj):
+        return obj.media_obj.meta_data.get("description")
+
+    @admin.display(description="Tags")
+    def tags(self, obj):
+        """Display tag names grouped by provider."""
+        if not obj.media_obj.tags:
+            return ""
+        tags_by_provider = {}
+        for tag in obj.media_obj.tags:
+            tags_by_provider.setdefault(tag["provider"], []).append(tag["name"])
+        return mark_safe(
+            "".join(
+                f"<p><strong>{provider}</strong>: {', '.join(names)}</p>"
+                for provider, names in tags_by_provider.items()
+            )
+        )
+
+    @admin.display(description="From")
+    def from_field(self, obj):
+        return f"provider: {obj.media_obj.provider}, source: {obj.media_obj.source}"
+
+    @admin.display(description="URLs")
+    def landing_urls(self, obj):
+        if not self.media_type or not obj:
+            return "N/A"
+        openverse_url = (
+            f"https://openverse.org/{self.media_type}/{obj.media_obj.identifier}"
+        )
+        return mark_safe(
+            f"<p>{create_link(obj.media_obj.foreign_landing_url, obj.media_obj.provider)}, "
+            f'{create_link(openverse_url, "openverse.org")}'
+        )
+
+    @admin.display(description="Other reports")
+    def other_reports(self, obj):
+        """
+        Display a table of other reports for the same media object.
+        Cannot use the ``TabularInline`` class because it requires a Parent -> Child relationship.
+        """
+        if not self.media_type or not obj:
+            return ""
+        reports = (
+            self.model.objects.filter(media_obj__identifier=obj.media_obj.identifier)
+            .exclude(id=obj.id)
+            .order_by("created_at")
+        )
+        if not reports:
+            return ""
+        result = "<table><thead><tr><th>Date</th><th>Report reason</th><th>Status</th><th>Report link</th></tr></thead><tbody>"
+        for report in reports:
+            report_link_html = f'<a href="{reverse("admin:api_imagereport_change", args=[report.id])}">{report.id}</a>'
+            created_at = report.created_at.strftime("%Y-%m-%d %H:%M:%S")
+            report_row = f"<tr><td>{created_at}</td><td>{report.reason}</td><td>{report.status}</td><td>{report_link_html}</td></tr>"
+            result += report_row
+        result += "</tbody></table>"
+        return mark_safe(result)
```

Comment:
This is really not the right way to do things in Django. Most important issue being that this string concatenation approach does not get any benefit from Django's template cache. A particularly important benefit, regarding maintainability, is the separation of view from model, which is essential for making the data model an understandable and distinctly separate thing from the presentation. Extracting all the presentation logic in the template forces us to write admin pages that cleanly silo the data model into template context, rather than creating functions like this where the two are meshed together such that it makes it unnecessarily harder to understand. That's even the more important benefit, in our context, is to avoid this approach because it is so much harder to read, reason about, and maintain than Django templates.

Please update the change_form template you've added in this PR to render the table in the template. It's also worth checking out Django's form building tools, if you're not familiar with them, as there are canonical ways to do just about everything: https://docs.djangoproject.com/en/5.0/topics/forms/#the-template and https://docs.djangoproject.com/en/5.0/ref/contrib/admin/#overriding-admin-templates (the latter of which you've almost definitely seen, given the template override you've already added, but including it just in case and for other reviewers who might not be familiar with Django, like Madison).